diff -ruN a/config.def.h b/config.def.h
--- a/config.def.h	2016-10-25 23:57:40.711061287 -0500
+++ b/config.def.h	2016-10-26 00:00:52.943065587 -0500
@@ -1,21 +1,24 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-static const unsigned int borderpx  = 1;        /* border pixel of windows */
+static const unsigned int borderpx  = 3;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
-static const char col_gray1[]       = "#222222";
-static const char col_gray2[]       = "#444444";
-static const char col_gray3[]       = "#bbbbbb";
-static const char col_gray4[]       = "#eeeeee";
-static const char col_cyan[]        = "#005577";
+static const char *fonts[]          = { "Liberation Mono:pixelsize=12:antialias=true:autohint=true" };
+static const char dmenufont[]       = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static const char S_base03[]        = "#002b36";
+static const char S_base02[]        = "#073642";
+static const char S_base01[]        = "#586e75";
+static const char S_base00[]        = "#657b83";
+static const char S_base0[]         = "#839496";
+static const char S_base1[]         = "#93a1a1";
+static const char S_base2[]         = "#eee8d5";
+static const char S_base3[]         = "#fdf6e3";
 static const char *colors[SchemeLast][3]      = {
-	/*               fg         bg         border   */
-	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
-	[SchemeSel] =  { col_gray4, col_cyan,  col_cyan  },
+	/*               fg       bg        border   */
+	[SchemeNorm] = { S_base0, S_base03, S_base01 },
+	[SchemeSel] =  { S_base1, S_base02, S_base3  },
 };
 
 /* tagging */
@@ -68,7 +71,7 @@
 
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
-static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
+static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", S_base03, "-nf", S_base0, "-sb", S_base02, "-sf", S_base1, NULL };
 static const char *termcmd[]  = { "st", NULL };
 static const char *dfindercmd[] = { "dfinder", NULL };
 static const char *dwmenucmd[] = { "dwmenu", NULL };
