diff -ruN a/config.def.h b/config.def.h
--- a/config.def.h	2016-10-25 23:53:27.079055613 -0500
+++ b/config.def.h	2016-10-25 23:56:25.292059600 -0500
@@ -19,22 +19,34 @@
 };
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = { "0-m", "1-e", "2-w", "3-o" };
 
 static const Rule rules[] = {
 	/* xprop(1):
 	 *	WM_CLASS(STRING) = instance, class
 	 *	WM_NAME(STRING) = title
 	 */
-	/* class      instance    title       tags mask     isfloating   monitor */
-	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	/* class      instance       title        tags mask   isfloating   monitor */
+	{ "Animate",  "animate",     NULL,        0,          True,        -1 },
+	{ "Display",  NULL,          NULL,        0,          True,        -1 },
+	{ "ffplay",   "ffplay",      NULL,	  0,   	      True,	   -1 },
+	{ "Xmessage", NULL,	     NULL,	  0,	      True,	   -1 },
+	{ "Emacs",    NULL,          NULL,        1 << 1,     False,       -1 },
+	{ "Firefox",  "Browser",     NULL,        1 << 2,     True,        -1 },
+	{ "Firefox",  "Navigator",   NULL,        1 << 2,     False,       -1 },
+	{ "Conkeror", "Download",    NULL,        1 << 2,     True,	   -1 },
+	{ "Conkeror", "Addons",      NULL,        1 << 2,     True,	   -1 },
+	{ "Conkeror", "Navigator",   NULL,        1 << 2,     False,	   -1 },
+	{ "Vlc",      "vlc",	     NULL,	  0,	      True,	   -1 },
+	{ "Soffice",  "libreoffice", NULL, 	  1 << 3,     True,   	   -1 },
+	{ NULL,	      "libreoffice", NULL, 	  1 << 3,     False,   	   -1 },
+	{ NULL,       "st-256color", NULL,        1 << 3,     False,       -1 },	
 };
 
 /* layout(s) */
-static const float mfact     = 0.55; /* factor of master area size [0.05..0.95] */
+static const float mfact     = 0.618; /* factor of master area size [0.05..0.95] */
 static const int nmaster     = 1;    /* number of clients in master area */
-static const int resizehints = 1;    /* 1 means respect size hints in tiled resizals */
+static const int resizehints = 0;    /* 1 means respect size hints in tiled resizals */
 
 static const Layout layouts[] = {
 	/* symbol     arrange function */
@@ -44,7 +56,7 @@
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -58,18 +70,26 @@
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
 static const char *termcmd[]  = { "st", NULL };
+static const char *dfindercmd[] = { "dfinder", NULL };
+static const char *dwmenucmd[] = { "dwmenu", NULL };
+static const char *emacscmd[] = { "emacsclient", "-a=", "-c", NULL };
+/* Reserved */
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
-	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
+	{ MODKEY,                       XK_x,      spawn,          {.v = dmenucmd } },
+	{ MODKEY,                       XK_a,      spawn,          {.v = dwmenucmd } },
+	{ 0,                            0x1008ff1b, spawn,         {.v = dfindercmd } },
+	{ MODKEY,			XK_e,	   spawn,	   {.v = emacscmd } },
+	/* Reserved */
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
 	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
 	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
-	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
-	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
+	{ MODKEY|ShiftMask,             XK_j,      setmfact,       {.f = -0.05} },
+	{ MODKEY|ShiftMask,             XK_k,      setmfact,       {.f = +0.05} },
 	{ MODKEY,                       XK_Return, zoom,           {0} },
 	{ MODKEY,                       XK_Tab,    view,           {0} },
 	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
@@ -84,15 +104,10 @@
 	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
 	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
-	TAGKEYS(                        XK_1,                      0)
-	TAGKEYS(                        XK_2,                      1)
-	TAGKEYS(                        XK_3,                      2)
-	TAGKEYS(                        XK_4,                      3)
-	TAGKEYS(                        XK_5,                      4)
-	TAGKEYS(                        XK_6,                      5)
-	TAGKEYS(                        XK_7,                      6)
-	TAGKEYS(                        XK_8,                      7)
-	TAGKEYS(                        XK_9,                      8)
+	TAGKEYS(                        XK_0,                      0)
+	TAGKEYS(                        XK_1,                      1)
+	TAGKEYS(                        XK_2,                      2)
+	TAGKEYS(                        XK_3,                      3)
 	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
 };
 
