diff '--exclude=.git' -ruN a/config.def.h b/config.def.h
--- a/config.def.h	2018-06-28 13:54:20.897905778 -0500
+++ b/config.def.h	2018-06-28 14:39:22.469756346 -0500
@@ -97,6 +97,7 @@
 	TAGKEYS(                        XK_7,                      6)
 	TAGKEYS(                        XK_8,                      7)
 	TAGKEYS(                        XK_9,                      8)
+	{ MODKEY,                       XK_q,      quit,           {1} },
 	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
 };
 
diff '--exclude=.git' -ruN a/dwm.c b/dwm.c
--- a/dwm.c	2018-06-28 14:37:02.683764078 -0500
+++ b/dwm.c	2018-06-28 14:40:55.938751176 -0500
@@ -300,6 +300,7 @@
 };
 static Atom wmatom[WMLast], netatom[NetLast], xatom[XLast];
 static int running = 1;
+static int restart = 0;
 static Cur *cursor[CurLast];
 static Clr **scheme;
 static Display *dpy;
@@ -1408,6 +1409,8 @@
 void
 quit(const Arg *arg)
 {
+	if(arg->i)
+		restart = 1;
 	running = 0;
 }
 
@@ -2578,6 +2581,8 @@
 #endif /* __OpenBSD__ */
 	scan();
 	run();
+	if(restart)
+		execvp(argv[0], argv);
 	cleanup();
 	XCloseDisplay(dpy);
 	return EXIT_SUCCESS;
