--- dwm.c.orig	2015-01-30 19:22:03.694894353 -0600
+++ dwm.c	2015-01-30 19:22:11.335894638 -0600
@@ -317,6 +317,7 @@
 };
 static Atom wmatom[WMLast], netatom[NetLast], xatom[XLast];
 static Bool running = True;
+static Bool restart = False;
 static Cursor cursor[CurLast];
 static Display *dpy;
 static DC dc;
@@ -1488,6 +1489,8 @@
 
 void
 quit(const Arg *arg) {
+	if(arg->i)
+		restart = True;
 	running = False;
 }
 
@@ -2556,6 +2559,8 @@
 	setup();
 	scan();
 	run();
+	if(restart)
+		execvp(argv[0], argv);
 	cleanup();
 	XCloseDisplay(dpy);
 	return EXIT_SUCCESS;
