diff -ruN a/config.def.h b/config.def.h
--- a/config.def.h	2016-10-25 19:42:16.337252096 -0500
+++ b/config.def.h	2016-10-25 19:51:48.024215964 -0500
@@ -111,6 +111,7 @@
 	TAGKEYS(                        XK_1,                      1)
 	TAGKEYS(                        XK_2,                      2)
 	TAGKEYS(                        XK_3,                      3)
+	{ MODKEY,                       XK_q,      quit,           {1} },
 	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
 };
 
diff -ruN a/dwm.c b/dwm.c
--- a/dwm.c	2016-10-25 19:49:31.465224594 -0500
+++ b/dwm.c	2016-10-25 19:50:36.448220487 -0500
@@ -301,6 +301,7 @@
 };
 static Atom wmatom[WMLast], netatom[NetLast], xatom[XLast];
 static int running = 1;
+static int restart = 0;
 static Cur *cursor[CurLast];
 static Scm scheme[SchemeLast];
 static Display *dpy;
@@ -1439,6 +1440,8 @@
 void
 quit(const Arg *arg)
 {
+	if(arg->i)
+		restart = 1;
 	running = 0;
 }
 
@@ -2575,6 +2578,8 @@
 	setup();
 	scan();
 	run();
+	if(restart)
+		execvp(argv[0], argv);
 	cleanup();
 	XCloseDisplay(dpy);
 	return EXIT_SUCCESS;
