--- a/config.def.h	2016-06-10 17:44:13.002373452 -0500
+++ b/config.def.h	2016-06-10 19:06:39.724662735 -0500
@@ -50,6 +50,7 @@
 static const float mfact     = 0.618;/* factor of master area size [0.05..0.95] */
 static const int nmaster     = 1;    /* number of clients in master area */
 static const int resizehints = 0;    /* 1 means respect size hints in tiled resizals */
+static const float popuptermmfact = 0.15; /* factor of master area size for pop-up terminal height [0.05..0.95] */
 
 #include "grid.c"
 #include "deck.c"
@@ -90,6 +91,8 @@
 static const char *printwin[] = { "screenshot", "-m", "window", "-f", "window", NULL };
 static const char *emacscmd[] = { "emacsclient", "-a=", "-c", NULL };
 static const char *dwmenu[]   =	{ "dwmenu", NULL };
+static const char popuptermname[] = "pop-up";
+static const char *popuptermcmd[] = { "st", "-t", popuptermname, "-e", "screen", "-d", "-RR", NULL };
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
@@ -136,6 +139,7 @@
 	TAGKEYS(			XK_3,			   3)
  	{ MODKEY,			XK_q,	   quit,	   {1} },
 	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	{ MODKEY,                       XK_p,	   togglepopup,    {.v = popuptermcmd } },
 };
 
 /* button definitions */
--- a/dwm.c	2016-06-10 17:44:13.001373452 -0500
+++ b/dwm.c	2016-06-10 19:25:24.100728489 -0500
@@ -269,6 +269,7 @@
 static int xerrordummy(Display *dpy, XErrorEvent *ee);
 static int xerrorstart(Display *dpy, XErrorEvent *ee);
 static void zoom(const Arg *arg);
+static void togglepopup(const Arg *arg);
 
 /* variables */
 static Systray *systray = NULL;
@@ -309,6 +310,7 @@
 
 /* configuration, allows nested code to access above variables */
 #include "config.h"
+static unsigned int popuptag = 1 << LENGTH(tags);
 
 struct Pertag {
 	unsigned int curtag, prevtag; /* current and previous tag */
@@ -392,7 +394,7 @@
 		*h = bh;
 	if (*w < bh)
 		*w = bh;
-	if (resizehints || c->isfloating || !c->mon->lt[c->mon->sellt]->arrange) {
+	if(strcmp(c->name, popuptermname) && (resizehints || c->isfloating || !c->mon->lt[c->mon->sellt]->arrange)) {
 		/* see last two sentences in ICCCM 4.1.2.3 */
 		baseismin = c->basew == c->minw && c->baseh == c->minh;
 		if (!baseismin) { /* temporarily remove base dimensions */
@@ -1218,6 +1220,15 @@
 	           && (c->x + (c->w / 2) < c->mon->wx + c->mon->ww)) ? bh : c->mon->my);
 	c->bw = borderpx;
 
+	if(!strcmp(c->name, popuptermname)) {
+	    c->mon->tagset[c->mon->seltags] |= c->tags = popuptag;
+	    c->isfloating = True;
+	    c->h = c->mon->wh * popuptermmfact;
+		resize(c, c->mon->wx, c->mon->wy + c->mon->wh - c->h, c->mon->ww - 2 * c->bw, c->h - 2 * c->bw, False);
+	}
+	else
+	    c->tags &= TAGMASK;
+
 	wc.border_width = c->bw;
 	XConfigureWindow(dpy, w, CWBorderWidth, &wc);
 	XSetWindowBorder(dpy, w, scheme[SchemeNorm].border->pix);
@@ -2581,6 +2592,28 @@
 	pop(c);
 }
 
+void
+togglepopup(const Arg *arg) {
+	Client *c = NULL;
+	unsigned int found = 0;
+
+	for(c = selmon->clients; c && !(found = c->tags & popuptag); c = c->next);
+	if(found) {
+	    unsigned int newtagset = selmon->tagset[selmon->seltags] ^ popuptag;
+	    if(newtagset) {
+	        selmon->tagset[selmon->seltags] = newtagset;
+	        focus(NULL);
+	        arrange(selmon);
+	    }
+	    if(ISVISIBLE(c)) {
+	        focus(c);
+	        restack(selmon);
+	    }
+	}
+	else
+	    spawn(arg);
+}
+
 int
 main(int argc, char *argv[])
 {
